src/gt4py/gtscript.py

```diff
@@ -523,12 +523,13 @@ def __len__(self):
 
 # GTScript builtins: domain axes
 class Axis:
```

Comment:
I think reusing the `Axis` class as a kind of `ShiftedAxis` is conceptually wrong. I think it would be better to have to different classes because they express actually different things, but maybe it is overkill. Can you explain your motivation to use this strategy here?

Reply:
I used it for the simplicity, but sure, I can change it. Actually we only need it internally anyway so it doesn't need to be exposed in `gtscript.py`.